### PR TITLE
Update tests for fullscreen event targets (element or document)

### PR DESCRIPTION
**Pull request by foolip**
Follows https://github.com/whatwg/fullscreen/pull/90.

### CODE REVIEW

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/234757127)

# Firefox (nightly) #
Testing web-platform-tests at revision 2e09fc80691506a29f1d36e40bbea2377821ee4a
Using browser at version BuildID 20170521100319; SourceStamp 5b74bbf20e803e299790d266fc6ebf5d53b7a1b7
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6017/fullscreen/api/element-request-fullscreen-not-allowed.html">/fullscreen/api/element-request-fullscreen-not-allowed.html</a></summary>


|                               Subtest                                | Results |                                                    Messages                                                    |
|----------------------------------------------------------------------|---------|----------------------------------------------------------------------------------------------------------------|
|                                                                      | OK      |                                                                                                                |
| `Element#requestFullscreen() when not allowed to request fullscreen` | FAIL    | `assert_equals: event.target expected Element node <div id="log"></div> but got Document node with 2 children` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/234757127)

# Firefox (nightly) #
Testing web-platform-tests at revision 2e09fc80691506a29f1d36e40bbea2377821ee4a
Using browser at version BuildID 20170521100319; SourceStamp 5b74bbf20e803e299790d266fc6ebf5d53b7a1b7
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6017/fullscreen/api/element-request-fullscreen-not-allowed.html">/fullscreen/api/element-request-fullscreen-not-allowed.html</a></summary>


|                               Subtest                                | Results |                                                    Messages                                                    |
|----------------------------------------------------------------------|---------|----------------------------------------------------------------------------------------------------------------|
|                                                                      | OK      |                                                                                                                |
| `Element#requestFullscreen() when not allowed to request fullscreen` | FAIL    | `assert_equals: event.target expected Element node <div id="log"></div> but got Document node with 2 children` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/234757128)

# Sauce (safari) #
Testing web-platform-tests at revision 2e09fc80691506a29f1d36e40bbea2377821ee4a
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6017/fullscreen/api/element-request-fullscreen-not-allowed.html">/fullscreen/api/element-request-fullscreen-not-allowed.html</a></summary>


|                               Subtest                                | Results |                                                     Messages                                                    |
|----------------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------------------|
|                                                                      | OK      |                                                                                                                 |
| `Element#requestFullscreen() when not allowed to request fullscreen` | FAIL    | `div.requestFullscreen is not a function. (In 'div.requestFullscreen()', 'div.requestFullscreen' is undefined)` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/234757128)

# Sauce (safari) #
Testing web-platform-tests at revision 2e09fc80691506a29f1d36e40bbea2377821ee4a
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6017/fullscreen/api/element-request-fullscreen-not-allowed.html">/fullscreen/api/element-request-fullscreen-not-allowed.html</a></summary>


|                               Subtest                                | Results |                                                     Messages                                                    |
|----------------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------------------|
|                                                                      | OK      |                                                                                                                 |
| `Element#requestFullscreen() when not allowed to request fullscreen` | FAIL    | `div.requestFullscreen is not a function. (In 'div.requestFullscreen()', 'div.requestFullscreen' is undefined)` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/234757134)

# Chrome (unstable) #
Testing web-platform-tests at revision 59f2307b83717077bcf6a6cab5000c725d6d680d
Using browser at version 60.0.3100.0 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6017/fullscreen/api/element-request-fullscreen-not-allowed.html">/fullscreen/api/element-request-fullscreen-not-allowed.html</a></summary>


|                               Subtest                                | Results |                  Messages                 |
|----------------------------------------------------------------------|---------|-------------------------------------------|
|                                                                      | OK      |                                           |
| `Element#requestFullscreen() when not allowed to request fullscreen` | FAIL    | `div.requestFullscreen is not a function` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/234757134)

# Chrome (unstable) #
Testing web-platform-tests at revision 59f2307b83717077bcf6a6cab5000c725d6d680d
Using browser at version 60.0.3100.0 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6017/fullscreen/api/element-request-fullscreen-not-allowed.html">/fullscreen/api/element-request-fullscreen-not-allowed.html</a></summary>


|                               Subtest                                | Results |                  Messages                 |
|----------------------------------------------------------------------|---------|-------------------------------------------|
|                                                                      | OK      |                                           |
| `Element#requestFullscreen() when not allowed to request fullscreen` | FAIL    | `div.requestFullscreen is not a function` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/234757135)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 59f2307b83717077bcf6a6cab5000c725d6d680d
Using browser at version 14.14393
Starting 10 test iterations

**Comment by foolip** [comment]
@bobholt, the result comments are duplicated on this PR, is this a known issue?

**Comment by bobholt** [comment]
@foolip: Not a known issue. As far as I can tell, it hasn't happened before (or in the PR since). 

For this to have happened, prbuildbot would have had to been invoked twice within very short order (otherwise the comments would have existed, and the bot's check for existing comments would have caught them). I think we can see this born out in the matching timestamps in the duplicate comments. The only way I could explain that occurrence is if Travis somehow fired the webhook twice.

**Comment by foolip** [comment]
> The only way I could explain that occurrence is if Travis somehow fired the webhook twice.

Presumably that can happen? If the server is very slow to respond, maybe it retries after a while, but the first request actually reached the system as well and just didn't get a reply?

**Comment by bobholt** [comment]
> Presumably that can happen? If the server is very slow to respond, maybe it retries after a while, but the first request actually reached the system as well and just didn't get a reply?

I've seen Travis (rarely) respond twice to commands that are only invoked once in the build script, so it's not outside the realm of possibility to me that it could have fired 2 webhook `POST`s.

If it waited any appreciable time between the two webhooks, we should only have seen one comment, as the previous comments would have already been in place (and found and edited by the script).

That said, I doubt Travis does any kind of webhook retry. It seems very "fire and forget." It's hard to tell, though, because they don't provide a UI into sent webhooks like GitHub does.

**Comment by foolip** [comment]
@upsuper, would you mind reviewing this as well?

**Review by zcorpan — APPROVED** [review]
LGTM, but could have more coverage, in particular: if *element* is connected but its node document is not *document*, is not tested, is it?

File: fullscreen/api/element-request-fullscreen-and-move-to-iframe-manual.html [hunk 1]
```diff
@@ -15,8 +15,9 @@
   iframeDoc.onfullscreenerror = t.unreached_func("fullscreenerror event in iframe");
   document.onfullscreenchange = t.unreached_func("fullscreenchange event");
   document.onfullscreenerror = t.step_func_done(() => {
-    assert_equals(document.fullscreenElement, null);
-    assert_equals(iframeDoc.fullscreenElement, null);
+    assert_equals(document.fullscreenElement, null, "outer fullscreen element");
+    assert_equals(iframeDoc.fullscreenElement, null, "inner fullscreen element");
+    assert_equals(event.target, document, "event target");
```
Comment (foolip) [inline fullscreen/api/element-request-fullscreen-and-move-to-iframe-manual.html]: @zcorpan, this and the same bit in element-request-fullscreen-and-remove-manual.html tests the event targeting.

**Comment by upsuper** [comment]
Is there test for checking what happens when the fullscreen element or any of its ancestor is removed from the document?

**Comment by foolip** [comment]
No, I'm now updating tests under model/ for that.

**Comment by foolip** [comment]
No, I'm now updating tests under model/ for that.

**Review by upsuper — APPROVED** [review]

**Comment by zcorpan** [comment]
Thanks. LGTM

**Pull request merged**